### PR TITLE
thumbnail WebP as WebP

### PR DESCRIPTION
**Pull request by tastytea**
I'm testing it since about 3 months (on a single user server) and it works fine.

Closes: https://github.com/matrix-org/synapse/issues/11840
Signed-off-by: Ronny (tastytea) Gutbrod <tastytea@tastytea.de>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by clokep** [comment]
At https://github.com/matrix-org/synapse/pull/7586#issuecomment-636011128 @erikjohnston said:

> > On the other hand, it might make sense to generate webp previews for webp images. This one also works fine. What do you think?
>
> Given some devices can't render webp I think it makes sense for thumbnails to be in a more widely supported format.

I think in the (almost) 3 years since that comment [support for WebP has increased](https://caniuse.com/webp), back in May of 2020 Safari (on both iOS and macOS) didn't support WebP, where it looks like it is widely supported now. The only folks who might have an issue are people on macOS < 11 where transparency of WebP is not supported. This shouldn't be any worse than what we currently do of thumbnailing to JPEG! (Latest IE 11 and KaiOS Browser 2.5 also don't support WebP, but those are hardly used.)

tl;dr I think it is reasonable to support this, but I have a few questions:

* Do you know if our minimum version of pillow supports thumbnailing to WebP?
* Do we need to ensure any operating system libraries are installed to support WebP?

Can we now update the tests at `tests/rest/media/v1/test_media_storage.py` to have expected output? It seems that in #7586 it was not "deterministic" so was left out?

**Comment by tastytea** [comment]
> At [#7586 (comment)](https://github.com/matrix-org/synapse/pull/7586#issuecomment-636011128) @erikjohnston said:
> tl;dr I think it is reasonable to support this, but I have a few questions:
> 
> * Do you know if our minimum version of pillow supports thumbnailing to WebP?
> * Do we need to ensure any operating system libraries are installed to support WebP?

Looks like it is supported since [pillow 2.4.0](https://github.com/python-pillow/Pillow/blob/2.4.0/depends/install_webp.sh) from 2014 and it needs libwebp.

**Review by squahtx — COMMENTED** [review]
Looks good. Now that we're relying on libwebp, we should also:
* Update the installation instructions in docs/setup/installation.md
  https://github.com/matrix-org/synapse/blob/4eb2f4e02b83c90519a8a55c562d1fafab7caf0b/docs/setup/installation.md#platform-specific-prerequisites
   The package name is `libwebp-dev` for Debian/Ubuntu/Raspbian, `libwebp-devel` for CentOS/Fedora and `libwebp-devel` for `OpenSUSE`
* Add a check for webp support to synapse/rest/media/v1/\_\_init\_\_.py
   https://github.com/matrix-org/synapse/blob/develop/synapse/rest/media/v1/__init__.py

**Comment by clokep** [comment]
Cross-referencing matrix-org/matrix-spec#453.

**Comment by tastytea** [comment]
Updated instructions and added check; also fixed the zlib check error message while i was there.

File: synapse/rest/media/v1/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2014-2016 OpenMarket Ltd
+# Copyright 2014-2016,2023 OpenMarket Ltd
```
Comment (squahtx) [inline synapse/rest/media/v1/__init__.py:1]: We usually assign copyright to the matrix.org foundation now.
```suggestion
# Copyright 2014-2016 OpenMarket Ltd
# Copyright 2023 The Matrix.org Foundation C.I.C
```

**Review by squahtx — COMMENTED** [review]
Thanks for updating the docs so thoroughly!

Just one minor thing left.

**Comment by tastytea** [comment]
updated since the location of `synapse/rest/media/v1/thumbnailer.py` changed to `synapse/media/thumbnailer.py`.
